sycl/test-e2e/ClangOffloadWrapper/no-sycl-rdc.cpp

```diff
@@ -0,0 +1,63 @@
+// Test clang-offload-wrapper with -fno-sycl-rdc
+
+// REQUIRES: gpu && level_zero
+// UNSUPPORTED: cuda, hip
+
+// RUN: %clangxx -fsycl --offload-new-driver %s %S/Inputs/add.cpp %S/Inputs/sub.cpp -o %t1.out \
+// RUN:  -Xdevice-post-link "-split=auto -emit-param-info -symbols -emit-exported-symbols -split-esimd -lower-esimd -O2 -spec-const=native" -v 2>&1 \
+// RUN: | FileCheck --check-prefix=CHECK-RDC %s
+
+// RUN: %{run} %t1.out
+
+// RUN: %clangxx -fsycl -fno-sycl-rdc --offload-new-driver %s %S/Inputs/add.cpp %S/Inputs/sub.cpp -o %t2.out \
+// RUN:  -Xdevice-post-link "-split=auto -emit-param-info -symbols -emit-exported-symbols -split-esimd -lower-esimd -O2 -spec-const=native" -v 2>&1 \
+// RUN: | FileCheck --check-prefix=CHECK-NO-RDC %s
+
+// RUN: %{run} %t2.out
+
+// Should only be one call of sycl-post-link for RDC mode.
+// CHECK-RDC: sycl-post-link{{.*}} -split=auto
+// CHECK-RDC-NOT: sycl-post-link
+
+// Should call sycl-post-link 3 times, once per each input file
+// CHECK-NO-RDC-COUNT-3: sycl-post-link{{.*}} -split=auto
+// CHECK-NO-RDC-NOT: sycl-post-link
+
+#include "sycl/sycl.hpp"
+#include <iostream>
+
+void add(sycl::queue q, int *result, int a, int b);
+void sub(sycl::queue q, int *result, int a, int b);
+
+int main() {
+  sycl::queue q;
+  int *result = sycl::malloc_host<int>(2, q);
```

Comment:
Technically, this needs `// REQUIRES: aspect_usm_host_allocations`, but the same would be true for any kind of USM and I'm not sure we're always checking for that. Maybe it's been changed recently because I thought that `malloc_device` had to be available in the past... Or maybe I just imagined that.

Reply:
Ill add that just to be safe, thanks